File: hw/aimacana/u2/hw18mongodb/typescript/src/database/connection.ts
import { MongoClient, Db } from 'mongodb';

export class MongoDBConnection {
    private static client: MongoClient | null = null;
    private static db: Db | null = null;
    private static readonly CONNECTION_STRING = 'mongodb://mongodb:mongodb@157.137.223.54:27017/mongodb?authSource=mongodb';
    private static readonly DATABASE_NAME = 'mongodb';

    private constructor() {}

    public static async getDatabase(): Promise<Db> {
        if (!this.db) {
            this.client = new MongoClient(this.CONNECTION_STRING, {
                serverSelectionTimeoutMS: 3000
            });
            await this.client.connect();
            this.db = this.client.db(this.DATABASE_NAME);
        }
        return this.db;
    }

    public static async close(): Promise<void> {
        if (this.client) {
            await this.client.close();
            this.client = null;
            this.db = null;
        }
    }
}

File: hw/aimacana/u2/hw18mongodb/python/database/connection.py
import pymongo

class MongoDBConnection:
    _client = None
    _db = None
    CONNECTION_STRING = "mongodb://mongodb:mongodb@157.137.223.54:27017/mongodb?authSource=mongodb"
    DATABASE_NAME = "mongodb"

    @classmethod
    def get_database(cls):
        if cls._db is None:
            cls._client = pymongo.MongoClient(cls.CONNECTION_STRING, serverSelectionTimeoutMS=3000)
            cls._db = cls._client[cls.DATABASE_NAME]
        return cls._db

    @classmethod
    def close(cls):
        if cls._client is not None:
            cls._client.close()
            cls._client = None
            cls._db = None
